internal/webhook/workloadmutation/webhookhandler.go

```diff
@@ -0,0 +1,98 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package workloadmutation contains the webhook that injects annotations into daemon-sets, deployments and stateful-sets.
+package workloadmutation
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"net/http"
+
+	"github.com/go-logr/logr"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	"github.com/aws/amazon-cloudwatch-agent-operator/internal/config"
+	"github.com/aws/amazon-cloudwatch-agent-operator/pkg/instrumentation/auto"
+)
+
+// +kubebuilder:webhook:path=/mutate-v1-workload,mutating=true,failurePolicy=ignore,groups="apps",resources=daemonsets;deployments;statefulsets,verbs=create;update,versions=v1,name=mworkload.kb.io,sideEffects=none,admissionReviewVersions=v1
+// +kubebuilder:rbac:groups="apps",resources=daemonsets;deployments;statefulsets,verbs=get;list;watch
+
+var _ WebhookHandler = (*workloadMutationWebhook)(nil)
+
+// WebhookHandler is a webhook handler that analyzes new daemon-sets and injects appropriate annotations into it.
+type WebhookHandler interface {
+	admission.Handler
+}
+
+// the implementation.
+type workloadMutationWebhook struct {
+	client             client.Client
+	decoder            *admission.Decoder
+	logger             logr.Logger
+	config             config.Config
+	annotationMutation *auto.AnnotationMutators
+}
+
+// NewWebhookHandler creates a new WorkloadWebhookHandler.
+func NewWebhookHandler(cfg config.Config, logger logr.Logger, decoder *admission.Decoder, cl client.Client, annotationMutation *auto.AnnotationMutators) WebhookHandler {
+	return &workloadMutationWebhook{
+		config:             cfg,
+		decoder:            decoder,
+		logger:             logger,
+		client:             cl,
+		annotationMutation: annotationMutation,
+	}
+}
+
+func (p *workloadMutationWebhook) Handle(ctx context.Context, req admission.Request) admission.Response {
+	if p.annotationMutation == nil {
+		return admission.Errored(http.StatusBadRequest, errors.New("failed to unmarshal annotation config"))
+	}
+
+	var err error
+	var marshaledObject []byte
+	var object runtime.Object
+	switch objectKind := req.Kind.Kind; objectKind {
+	case "DaemonSet":
```

Comment:
Can we look for consts for these from the k8s lib?

Reply:
I was looking for this too. Wasn't able to find it.

Reply:
Couldnt find constants, we could declare our own